system-addon/lib/SectionsManager.jsm

```diff
@@ -77,18 +85,34 @@ const SectionsManager = {
     this.updateSection(id, {enabled: false, rows: []}, true);
     this.emit(this.DISABLE_SECTION, id);
   },
+  updateSections() {
+    this.sections.forEach((section, id) => this.updateSection(id, section, true));
+  },
   updateSection(id, options, shouldBroadcast) {
+    this.updateSectionContextMenuOptions(options);
+
     if (this.sections.has(id)) {
       this.sections.set(id, Object.assign(this.sections.get(id), options));
       this.emit(this.UPDATE_SECTION, id, options, shouldBroadcast);
     }
   },
+  updateSectionContextMenuOptions(options) {
+    if (options.availableContextMenuOptions) {
+      options.contextMenuOptions = options.availableContextMenuOptions.filter(
+        o => !this.CONTEXT_MENU_PREFS[o] || Services.prefs.getBoolPref(this.CONTEXT_MENU_PREFS[o]));
```

Comment:
Add a comment here or for the whole method that this is to show the option unless there's a pref to conditionally show the option.